README.md

```diff
@@ -48,63 +59,146 @@ python scripts/make.py "qt/bin" --lib-path "nomacs/3rd-party/build"
 - check if your Qt is set correctly (otherwise set the path to `qt_install_dir/qtbase/bin/qmake.exe`)
 - check if your builds proceeded correctly
 
-## Build nomacs (Ubuntu)
+## Build nomacs (Linux/Unix)
 
-Get the required packages:
+Before you build nomacs, please note the following:
 
-- For Ubuntu 21.04 and above:
+- We recommend using the Qt6 version of nomacs. However, you may want to match the Qt version of your desktop environment (e.g. KDE in Ubuntu 24.04 is Qt5-based).
+- [kimageformats-plugins]([https://github.com/KDE/kimageformats](https://invent.kde.org/frameworks/kimageformats)) is an optional dependency that provides additional formats such as AVIF, HEIC/HEIF, and JPEG XL/JXL. The Qt version of the plugins should match the Qt version when compiling nomacs.
+- Zip file support requires Quazip, which has varied support in distributions. If the system package is missing or does not work, you can set `USE_SYSTEM_QUAZIP=NO` to use the version in nomacs/3rdparty. However, you may also need to remove the any system quazip development package temporarily. (such as `llibquazip*-dev` for Ubuntu)
 
-  ```console
-  sudo apt-get install debhelper cdbs qt5-qmake qttools5-dev-tools qtbase5-dev qttools5-dev libqt5svg5-dev qt5-image-formats-plugins libexiv2-dev libraw-dev libopencv-dev cmake libtiff-dev libquazip5-dev libwebp-dev git build-essential lcov libzip-dev
-  ```
+### Get the required packages
 
-- For older Ubuntu versions:
+The package lists and cmake configuration listed below enable all features in nomacs, which we recommend. However, if you do not want a particular feature you can omit certain packages.
 
-  ```console
-  sudo apt-get install debhelper cdbs qt5-qmake qttools5-dev-tools qt5-default qttools5-dev libqt5svg5-dev qt5-image-formats-plugins libexiv2-dev libraw-dev libopencv-dev cmake libtiff-dev libquazip5-dev libwebp-dev git build-essential lcov libzip-dev
-  ```
+This is the list of configurable packages and what they provide in nomacs. If you want to be sure a feature is included/excluded set the cmake option for that feature. By default, all features are enabled if the dependencies are met**.
 
-Clone the nomacs repository from GitHub:
+| Package | Required | Depends | cmake | Description
+| -       | -        | -       | -     | -
+| Qt      |  Yes     |         | `QT_VERSION_MAJOR` | Qt components: Core, Concurrent, Network, PrintSupport, SVG, Widgets, Core5Compat
+| LibRAW  |  No      | OpenCV  | `ENABLE_RAW`       | Enables reading RAW images
+| LibTiff |  No      | OpenCV  | `ENABLE_TIFF`      | Enable reading multi-page TIFF
```

Comment:
Do these two libraries depend on OpenCV?

Reply:
LibRAW definitely does. LibTiff I'm not sure, it was not clear from scanning the code for `WITH_LIBTIFF`, however it is a cmake dependency. Maybe that is outdated.

Reply:
Could you confirm? The libraw packages in Ubuntu and Arch also do not list opencv as a dependency.

https://www.libraw.org/docs/Install-LibRaw.html
![image](https://github.com/user-attachments/assets/f6b6f8c9-f6bf-4abb-ae75-b7b656053d78)

Reply:
To clarify, the dependency is in nomacs itself, not the library. In otherwords, OpenCV is integral to how we use LibRAW.

As for tiff,

`cmake -DENABLE_OPENCV=OFF -DENABLE_RAW=OFF -DENABLE_TIFF=ON  ../ImageLounge/`

Does not work; however it might be made to work easily. I did not check functionality, but simply removing the check from Unix.cmake and the includes in DkBasicLoader.cpp allowed it to compile. Now, for the page extraction plugin, which I believe takes advantage of multi-page TIFF support, that might be another story.

Reply:
>To clarify, the dependency is in nomacs itself, not the library. In otherwords, OpenCV is integral to how we use LibRAW.

The current wording did not look like that. I would probably invert the table if that is the intent.

```md
#### Build options and their dependencies

The following Qt components are necessary: Core, Concurrent, Network, PrintSupport, SVG, Widgets, Core5Compat.

There are other optional features that can be enabled during build:

| Option             | Depends on     | Requires        | Description
| ------------------ | -------------- | --------------- | -
| `ENABLE_RAW`       | LibRAW         | `ENABLE_OPENCV` | Enables reading RAW images
| `ENABLE_TIFF`      | LibTiff        | `ENABLE_OPENCV` | Enable reading multi-page TIFF
| `ENABLE_QUAZIP`    | Quazip**       |                 | Enable reading from zip files
| `ENABLE_OPENCV`    | OpenCV         |                 | RAW, TIFF, Adjustments, High-quality thumbnails, DRIF files, histogram, mosaic
| `ENABLE_PLUGINS`   | nomacs/plugins | `ENABLE_OPENCV` | Paint on image, Composite, Affine Transform, Fake Miniatures, Page Extraction
| `ENABLE_TESTING`   | GoogleTest     |                 | Enables `make check` target for unit testing

#### Runtime dependencies

- Corresponding libraries for the enabled options.
- QImageFormats: Enables reading ICNS, MNG, TGA, TIFF, WBMP, WEBP
- KImageFormats: Enables reading AVIF, HEIF/HEIC, JXL, EXR, EPS and [more](https://github.com/KDE/kimageformats)

```